apps/bottle/src/store/query/ServerFetchBoundary.tsx

```diff
@@ -0,0 +1,20 @@
+import { type FetchQueryOptions, HydrationBoundary, dehydrate } from '@tanstack/react-query';
+import type { ReactNode } from 'react';
+import { getQueryClient } from './getQueryClient';
+
+export type FetchOptions = Pick<FetchQueryOptions, 'queryKey' | 'queryFn'>;
+
+type Props = {
+  fetchOptions: FetchOptions[] | FetchOptions;
+  children: ReactNode | ReactNode[];
+};
+
+export async function ServerFetchBoundary({ fetchOptions, children }: Props) {
+  const queryClient = getQueryClient();
+
+  Array.isArray(fetchOptions)
+    ? Promise.all(fetchOptions.map(prefetchOption => queryClient.fetchQuery(prefetchOption)))
+    : queryClient.fetchQuery(fetchOptions);
```

Comment:
@stakbucks 
I'm curious why you changed it from prefetchQuery to fetchQuery.

Reply:
I actually explained this in the PR body.
This is because prefetchQuery does not throw errors. So errors are not caught by the Next.js's internal ErrorBoundary.
-> see: https://tanstack.com/query/latest/docs/reference/QueryClient/#queryclientfetchquery

Reply:
"prefetchQuery is an asynchronous method that can be used to prefetch a query before it is needed or rendered with useQuery and friends. The method works the same as fetchQuery except that it will not throw or return any data"

Reply:
I am also planning to add a custom ErrorBoundary to handle mutation errors. Maybe you can help me implement Suspensive